chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -369,11 +372,18 @@ pub fn pre_validate_chunk_state_witness(
 
     let current_protocol_version =
         epoch_manager.get_epoch_protocol_version(&state_witness.epoch_id)?;
-    if !checked_feature!(
+    let transaction_validity = if checked_feature!(
         "protocol_feature_relaxed_chunk_validation",
         RelaxedChunkValidation,
         current_protocol_version
     ) {
+        let mut check = chain.transaction_validity_check(last_chunk_block.header().clone());
```

Comment:
`last_chunk_block` is not the block we should use here.
on the chunk producer side `prepare_transactions` is called with:
```&mut chain.transaction_validity_check(prev_block.header().clone())```
so we should use parent block the way we do in `validate_prepared_transactions`:
```let parent_block = chain.chain_store().get_block(chunk_header.prev_block_hash())?;```

I suggest adding parent block to `StateWitnessBlockRange`, it seems like it belongs there